frame/transaction-storage/src/mock.rs

```diff
@@ -0,0 +1,148 @@
+		{
+			System: frame_system::{Pallet, Call, Config, Storage, Event<T>},
+			Balances: pallet_balances::{Pallet, Call, Config<T>, Storage, Event<T>},
+			TransactionStorage: pallet_transaction_storage::{
+				Pallet, Call, Storage, Config<T>, Inherent, Event<T>
+			},
+		}
+	);
+
+	parameter_types! {
+		pub const BlockHashCount: u64 = 250;
+		pub const SS58Prefix: u8 = 42;
+	}
+
+	impl frame_system::Config for Test {
+		type BaseCallFilter = ();
+		type BlockWeights = ();
+		type BlockLength = ();
+		type Origin = Origin;
+		type Call = Call;
+		type Index = u64;
+		type BlockNumber = u64;
+		type Hash = H256;
+		type Hashing = BlakeTwo256;
+		type AccountId = u64;
+		type Lookup = IdentityLookup<Self::AccountId>;
+		type Header = Header;
+		type Event = Event;
+		type BlockHashCount = BlockHashCount;
+		type DbWeight = ();
+		type Version = ();
+		type PalletInfo = PalletInfo;
+		type AccountData = pallet_balances::AccountData<u64>;
+		type OnNewAccount = ();
+		type OnKilledAccount = ();
+		type SystemWeightInfo = ();
+		type SS58Prefix = SS58Prefix;
+		type OnSetCode = ();
+	}
+
+	parameter_types! {
+		pub const ExistentialDeposit: u64 = 1;
+	}
+
+	impl pallet_balances::Config for Test {
+		type Balance = u64;
+		type DustRemoval = ();
+		type Event = Event;
+		type ExistentialDeposit = ExistentialDeposit;
+		type AccountStore = System;
+		type WeightInfo = ();
+		type MaxLocks = ();
+	}
+
+	parameter_types! {
+		pub IgnoredIssuance: u64 = Balances::total_balance(&0); // Account zero is ignored.
+		pub const QueueCount: u32 = 3;
+		pub const MaxQueueLen: u32 = 3;
+		pub const FifoQueueLen: u32 = 1;
+		pub const Period: u64 = 3;
+		pub const MinFreeze: u64 = 2;
+		pub const IntakePeriod: u64 = 2;
+		pub const MaxIntakeBids: u32 = 2;
+	}
+
+	impl pallet_transaction_storage::Config for Test {
+		type Event = Event;
+		type Call = Call;
+		type Currency = Balances;
+		type WeightInfo = ();
+	}
+
+	pub fn new_test_ext() -> sp_io::TestExternalities {
+		let t = GenesisConfig {
+			frame_system: Default::default(),
+			pallet_balances: pallet_balances::GenesisConfig::<Test> {
+				balances: vec![(1, 1000000000), (2, 100), (3, 100), (4, 100)]
+			},
+			pallet_transaction_storage: pallet_transaction_storage::GenesisConfig::default(),
+		}.build_storage().unwrap();
+		t.into()
+	}
+
+}
+
+pub fn run_to_block<T: Config>(n: u32) {
+	while frame_system::Pallet::<T>::block_number() < (n as u32).into() {
+		pallet_transaction_storage::Pallet::<T>::on_finalize(frame_system::Pallet::<T>::block_number());
+		frame_system::Pallet::<T>::on_finalize(frame_system::Pallet::<T>::block_number());
+		frame_system::Pallet::<T>::set_block_number(frame_system::Pallet::<T>::block_number() + One::one());
+		frame_system::Pallet::<T>::on_initialize(frame_system::Pallet::<T>::block_number());
+		pallet_transaction_storage::Pallet::<T>::on_initialize(frame_system::Pallet::<T>::block_number());
+	}
+}
+
+pub fn setup<T: Config>() -> Result<(), &'static str> {
```

Comment:
This should just be part of your `new_test_ext`?

Reply:
also unclear why be generic over `T` here.

Reply:
Okay -- I see that you are using this in benchmarks as well. Makes sense. 

But note that I still think it is a bad pattern. You are now overriding the fee to `2` in all benchmarks, regardless of it being test or the real benchmarks. In fact, in real benchmarks (i.e. run against `substrate-node` or `polkadot`) `2` as balance type is pretty much insubstantial and below ED. 

I think the correct thing is that you should not manually setup these things in your benchmarks. Then, 

1. For testing benchmarks, put them in `new_test_ext()`
2. For real benchmarks, assume that they chain that is running the benchmarks already has sensible value. You can assert to make sure the values are sane though (`assert!(fee > ED)` etc).

Reply:
Done